helpers/smtptls/smtptls.go

```diff
@@ -0,0 +1,52 @@
+package smtptls
+
+import (
+	"crypto/tls"
+	"net/smtp"
+
+	"github.com/pkg/errors"
+)
+
+func Connect(host string, tlsc tls.Config) (*smtp.Client, error) {
+	client, err := smtp.Dial(host)
+	if err != nil {
+		return nil, errors.Wrap(err, "dialing smtp")
+	}
+	if ok, _ := client.Extension("STARTTLS"); ok {
+		if err = client.StartTLS(&tlsc); err != nil {
+			return nil, errors.Wrap(err, "starting tls")
+		}
+	} else {
+		return nil, errors.Errorf("%s doesn't advertise STARTTLS support", host)
+	}
+
+	return client, nil
+}
+
+func SendMail(c *smtp.Client, from string, to []string, msg []byte) error {
+	if err := c.Mail(from); err != nil {
+		return errors.Wrap(err, "sending MAIL FROM")
+	}
+
+	if err := c.Rcpt(to[0]); err != nil {
```

Comment:
Not a big issue but I would probably lean towards checking array list size and non nil before accessing index 0. It will fail if it is not right but looks like this code is trying to provide helpful details about the root cause of an error.

Reply:
Addressed above, thanks!